### PR TITLE
Terminate timer scheduler thread during shutdown

### LINKED ISSUE #16153: Terminate timer scheduler thread during shutdown

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #16153 

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16184)

### CODE REVIEW

File: components/constellation/timer_scheduler.rs [hunk 1]
```diff
@@ -105,7 +106,15 @@ impl TimerScheduler {
...
+                    let mut shutting_down = false;
+                    match req {
+                        TimerSchedulerMsg::Exit => shutting_down = true,
+                        _ => {}
+                    }
                     let _ = req_sender.send(req);
+                    if shutting_down {
+                        break;
+                    }
                     timeout_thread.unpark();
```
Comment (jdm) [inline components/constellation/timer_scheduler.rs]: We should probably unpark the thread before breaking out of this loop.

**Review by jdm — CHANGES_REQUESTED** [review]
This looks really good!

**Comment by jdm** [comment]
Looks good. Squash and r=me!
@bors-servo: delegate+

**Review by jdm — APPROVED** [review]

**Comment by ferjm** [comment]
Thank you Josh!

@bors-servo r=jdm

**Pull request merged**